src/main/java/ca/fxco/pistonlib/pistonLogic/structureResolvers/BasicStructureResolver.java

```diff
@@ -165,20 +164,23 @@ protected static boolean canAdjacentBlockStick(Direction dir, BlockState state,
 
     // Stickiness checks
     protected static boolean canAdjacentBlockStick(Direction dir, BlockState state, BlockState adjState, boolean attemptIndirect) {
-        BlockStateBaseExpandedSticky stick = (BlockStateBaseExpandedSticky)adjState;
-        if (stick.usesConfigurablePistonStickiness()) {
-            if (!stick.isSticky()) {
+        BlockStateBaseExpandedSticky adjStick = (BlockStateBaseExpandedSticky)adjState;
+        if (adjStick.usesConfigurablePistonStickiness()) {
+            if (!adjStick.isSticky()) {
                 return attemptIndirect && canIndirectBlockStick(dir, state, adjState);
             }
-            StickyType type = stick.sideStickiness(dir.getOpposite());
-            if (type == StickyType.CONDITIONAL && !type.canStick(state, adjState, dir)) {
-                return true;
+            StickyType type = adjStick.sideStickiness(dir.getOpposite());
+            if (type == StickyType.CONDITIONAL) {
+                if (type.canStick(state, adjState, dir)) {
+                    return true;
+                }
+                return attemptIndirect && canIndirectBlockStick(dir, state, adjState);
             }
             return type != StickyType.NO_STICK; // If NO_STICKY, even indirect sticky cannot stick to this side!
```

Comment:
shouldn't this be 
`if (type == StickyType.NO_STICK) return false`
otherwise blocks which uses configurable piston stickiness and has non NO_STICK type will stick